ansible/roles/atmo-vnc/tasks/xauth.yml

```diff
@@ -0,0 +1,12 @@
+---
+
+- block:
+    - name: Get xauth cookies
+      shell: "xauth list | grep :{{ item }} | awk '{print $3}'"
+      register: cookies
+
+    - name: Re-add cookies with new hostname
+      shell: 'xauth add "$(hostname)/unix:{{ item }}" MIT-MAGIC-COOKIE-1 {{ cookies.stdout_lines[0] }}'
+      when: cookies is defined and cookies.stdout_lines|length > 0
+  become: yes
+  become_user: '{{ ATMOUSERNAME }}'
```

Comment:
If you look at the man page, just below "add" there is a "generate" command. I think the approach can be altered to: for each display, generate the cookie for that display.

Reply:
I squashed it all to one commit with a better message (in my opinion). I tried out using generate and could only get it to work if xauth was completely empty. So after generating for `:1`, it doesn't work for `:2` and `:5`. I'm looking to find out why this is happening but I don't see much